### PR TITLE
Do not manage brokers through the Organization API 

### LINKED ISSUE #29268: Do not manage brokers through the Organization API
### Description

Similarly to organization members, the Organization API should only link existing brokers to organizations so that creating, updating, and removing brokers in a realm is only possible through the Admin Identity Provider API.

The Organization API should only link, return a representation, and unlink brokers from an organization.

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #29268 

Depends on for https://github.com/keycloak/keycloak/pull/29064.

### CODE REVIEW

File: server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java [hunk 1]
```diff
@@ -1645,4 +1650,39 @@ private static <T> Set<T> collectionToSet(Collection<T> collection) {
...
+            return;
+        }
+
+        IdentityProviderModel existing = realm.getIdentityProvidersStream()
+                .filter(p -> Objects.equals(p.getAlias(), representation.getAlias()) || Objects.equals(p.getInternalId(), representation.getInternalId()))
+                .findFirst().orElse(null);
+
+        if (existing == null) {
+            return;
+        }
```
Comment (sguilhen) [inline server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java:1665]: Why return here? If you are importing an IDP, don't you want to check if the representation matches an org (potentially) and one of its domain regardless of it existing already?

File: services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProvidersResource.java [hunk 2]
```diff
@@ -102,37 +96,40 @@ public Stream<IdentityProviderRepresentation> getIdentityProviders() {
...
+    @DELETE
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response delete(@PathParam("alias") String alias) {
+        IdentityProviderModel broker = realm.getIdentityProviderByAlias(alias);
 
-                if (domain != null && organization.getDomains().map(OrganizationDomainModel::getName).noneMatch(domain::equals)) {
-                    return Response.status(Status.BAD_REQUEST).build();
-                }
+        if (!isOrganizationBroker(broker)) {
+            throw ErrorResponse.error("Identity provider not found with the given alias", Status.NOT_FOUND);
```
Comment (sguilhen) [inline services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProvidersResource.java:118]: Shouldn't the message indicate that no provider with the specified alias is currently bound to the org? If you prefer we can create a follow up issue to revisit all messages returned from endpoints.

Comment (pedroigor) [reply services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProvidersResource.java:118]: Changed.

**Review by sguilhen — APPROVED** [review]

**Review by mhajas — APPROVED** [review]
Approving based on the review from @sguilhen

**Pull request merged**